browser/typescript/v2/react-app/src/ampli/index.ts

```diff
@@ -0,0 +1,829 @@
+   *
+   * Description for event with enum types
+   *
+   * Owner: Test codegen
+   *
+   * @param properties The event's properties (e.g. optional enum)
+   * @param options Amplitude event options.
+   */
+  eventWithEnumTypes(
+    properties: EventWithEnumTypesProperties,
+    options?: EventOptions,
+  ) {
+    return this.track(new EventWithEnumTypes(properties), options);
+  }
+
+  /**
+   * Event With Optional Array Types
+   *
+   * [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20With%20Optional%20Array%20Types)
+   *
+   * Description for event with optional array types
+   *
+   * Owner: Test codegen
+   *
+   * @param properties The event's properties (e.g. optionalBooleanArray)
+   * @param options Amplitude event options.
+   */
+  eventWithOptionalArrayTypes(
+    properties?: EventWithOptionalArrayTypesProperties,
+    options?: EventOptions,
+  ) {
+    return this.track(new EventWithOptionalArrayTypes(properties), options);
+  }
+
+  /**
+   * Event With Optional Properties
+   *
+   * [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20With%20Optional%20Properties)
+   *
+   * Event w optional properties description
+   *
+   * Owner: Test codegen
+   *
+   * @param properties The event's properties (e.g. optionalArrayNumber)
+   * @param options Amplitude event options.
+   */
+  eventWithOptionalProperties(
+    properties?: EventWithOptionalPropertiesProperties,
+    options?: EventOptions,
+  ) {
+    return this.track(new EventWithOptionalProperties(properties), options);
+  }
+
+  /**
+   * Event With Template Properties
+   *
+   * [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20With%20Template%20Properties)
+   *
+   * Event with template properties description
+   *
+   * Owner: Test codegen
+   *
+   * @param properties The event's properties (e.g. optional_event_property)
+   * @param options Amplitude event options.
+   */
+  eventWithTemplateProperties(
+    properties: EventWithTemplatePropertiesProperties,
+    options?: EventOptions,
+  ) {
+    return this.track(new EventWithTemplateProperties(properties), options);
+  }
+}
+
+export const ampli = new Ampli();
+
+// BASE TYPES
+type ConfigExt = Partial<amplitude.Types.BrowserOptions>;
+
+export const SpecialEventType = amplitude.Types.SpecialEventType;
+
+export type BaseEvent = amplitude.Types.BaseEvent;
+
+export type IdentifyEvent = BaseEvent & { event_type: amplitude.Types.SpecialEventType.IDENTIFY };
+export type GroupEvent = BaseEvent & { event_type: amplitude.Types.SpecialEventType.GROUP_IDENTIFY };
+export type Event = amplitude.Types.Event;
+
+type BaseEventOptions = amplitude.Types.BaseEvent;
+export type EventOptions = BaseEventOptions;
+export type IdentifyOptions = BaseEventOptions;
+export type GroupOptions = BaseEventOptions;
```

Comment:
I think we can remove these types. There were placeholders in v1 incase we needed Identify/Group specific options. In v2 EventOptions works for all event types.